tests/graph_components/validators/test_config_validator.py

```diff
@@ -0,0 +1,805 @@
+        for featurizer in [
+            RegexFeaturizerGraphComponent,
+            RegexEntityExtractorGraphComponent,
+        ]
+    ],
+)
+def test_nlu_warn_if_lookup_table_is_not_used(
+    featurizer: Type[GraphComponent],
+    consumer: Type[GraphComponent],
+    warns_featurizer: bool,
+    warns_consumer: bool,
+):
+    training_data = TrainingData(
+        training_examples=[Message({})],
+        lookup_tables=[{"elements": "this-is-no-file-and-that-does-not-matter"}],
+    )
+    assert training_data.lookup_tables is not None
+    component_types = [WhitespaceTokenizerGraphComponent, featurizer, consumer]
+    component_types = [type for type in component_types if type is not None]
+
+    expected_warnings = []
+    if warns_featurizer:
+        warning = (
+            "You have defined training data consisting of lookup tables, "
+            "your NLU configuration does not include a featurizer using the "
+            "lookup table."
+        )
+        expected_warnings.append(warning)
+    if warns_consumer:
+        warning = (
+            "You have defined training data consisting of lookup tables, but "
+            "your NLU configuration does not include any components "
+            "that uses the features created from the lookup table. "
+        )
+        expected_warnings.append(warning)
+    _test_validation_warnings_with_default_configs(
+        training_data=training_data,
+        component_types=component_types,
+        warnings=expected_warnings,
+    )
+
+
+def test_nlu_warn_if_lookup_table_and_crf_extractor_pattern_feature_mismatch():
+    training_data = TrainingData(
+        training_examples=[Message({})],
+        lookup_tables=[{"elements": "this-is-no-file-and-that-does-not-matter"}],
+    )
+    assert training_data.lookup_tables is not None
+    importer = DummyImporter(training_data=training_data)
+
+    match = (
+        f"You have defined training data consisting of lookup tables, "
+        f"but your NLU configuration's "
+        f"'{CRFEntityExtractorGraphComponent.__name__}' does not include the "
+        f"'pattern' feature"
+    )
+
+    # without 'pattern'
+    crf_schema_node = SchemaNode(
+        {}, CRFEntityExtractorGraphComponent, "", "", {"features": [["pos"]]},
+    )
+    graph_schema = GraphSchema(
+        {
+            "tokenizer": SchemaNode({}, WhitespaceTokenizerGraphComponent, "", "", {}),
+            "featurizer": SchemaNode({}, RegexFeaturizerGraphComponent, "", "", {}),
+            "crf": crf_schema_node,
+        }
+    )
+    validator = DefaultV1ConfigValidator(graph_schema)
+    with pytest.warns(UserWarning, match=match):
+        validator.validate(importer)
+
+    # with 'pattern'
+    crf_schema_node.config = {"features": [["suffix1", "pattern"], ["pos"]]}
+    validator = DefaultV1ConfigValidator(graph_schema)
+    with pytest.warns(None) as records:
+        validator.validate(importer)
+        assert len(records) == 0
```

Comment:
This tests a different case - should we make it a separate unit test?

Reply:
Hm, I've left it in the same test_* because it's testing the same "if and only if". I can parametrize the test, though. Then we get the best of both worlds